modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -34,304 +39,208 @@ extension AddressBookClient: DependencyKey {
 
         return Self(
             allLocalContacts: {
-                // return latest known contacts
+                // return latest known contacts or load ones for the first time
                 guard latestKnownContacts == nil else {
-                    if let contacts = latestKnownContacts {
-                        return contacts
-                    } else {
-                        return .empty
-                    }
+                    return latestKnownContacts ?? .empty
                 }
 
                 // contacts haven't been loaded from the locale storage yet, do it
                 do {
                     guard let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else {
                         throw AddressBookClientError.documentsFolder
                     }
-                    let fileURL = documentsDirectory.appendingPathComponent(Constants.component)
-                    let encryptedContacts = try Data(contentsOf: fileURL)
 
-                    let decryptedContacts = try AddressBookClient.decryptData(encryptedContacts)
-                    latestKnownContacts = decryptedContacts
+                    // Try to find and get the data from the encrypted file with the latest encryption version
+                    let encryptedFileURL = documentsDirectory.appendingPathComponent(try AddressBookClient.filenameForEncryptedFile())
 
-                    return decryptedContacts
+                    if let contactsData = try? Data(contentsOf: encryptedFileURL) {
+                        // file exists, try to find the unencrypted file in case the delete failed
+                        let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
+                        if FileManager.default.fileExists(atPath: unencryptedFileURL.path) {
+                            try? FileManager.default.removeItem(at: unencryptedFileURL)
+                        }
+                        
+                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
```

Comment:
This doesn't follow the design:
> If the parse succeeds, also look for a file with the previous filename of the unencrypted address book format and delete it if present.

The attempt to delete the unencrypted file should occur only if the parse succeeds.
```suggestion
                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)

                        // The encrypted file exists, and can be decrypted and parsed. Try to find the
                        // unencrypted file and delete it, in case an earlier delete failed.
                        let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
                        if FileManager.default.fileExists(atPath: unencryptedFileURL.path) {
                            try? FileManager.default.removeItem(at: unencryptedFileURL)
                        }
```

Reply:
Swapped the order of functions, attempt to delete is now performed only if parse passed.